packages/client/src/components/HTMLResourceRenderer.tsx

```diff
@@ -85,3 +102,21 @@ export const HTMLResourceRenderer = ({
 };
 
 HTMLResourceRenderer.displayName = 'HTMLResourceRenderer';
+
+function postResponse(
+  uiActionResult: UIActionResult,
+  event: MessageEvent,
+  type: (typeof InternalMessageType)[keyof typeof InternalMessageType],
+  payload?: Record<string, unknown>,
+) {
+  if (uiActionResult.messageId) {
+    event.source?.postMessage({
+      type,
+      payload: {
+        messageId: uiActionResult.messageId,
+        originalMessage: event.data,
+        ...(payload ?? {}),
```

Comment:
Typically, I would expect the partner's value to not be merged with other metadata; I would expect the `messageId` and any other field like that to just be on the "envelope", separate from the data itself. This approach will also fall over if the `onUiAction` returns anything other than an object, which we don’t seem to be validating at the callsite (and which seems a little restrictive?)

Reply:
Good point about the payload destructuring, I changed it to simply put the response field (keeping the `{ type, payload }` standard structure of all of the other messages
```
type,
payload: {
  messageId: uiActionResult.messageId,
  response,
},
```